sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyWriterTest.java

```diff
@@ -193,6 +205,53 @@ public void timeout2() throws Exception {
         subscriber.assertError(RequestTimeoutException.class);
     }
 
+    @Test(groups = "unit", dataProvider = "storeResponseArgProvider")
+    public void storeResponseRecordedOnException(Exception ex, StoreResponse storeResponse) {
+        DiagnosticsClientContext clientContext = mockDiagnosticsClientContext();
+        TransportClientWrapper transportClientWrapper;
+
+        if (ex != null) {
+            transportClientWrapper = new TransportClientWrapper.Builder.ReplicaResponseBuilder
+                .SequentialBuilder()
+                .then(ex)
+                .build();
+        } else {
+            transportClientWrapper = new TransportClientWrapper.Builder.ReplicaResponseBuilder
+                .SequentialBuilder()
+                .then(storeResponse)
+                .build();
+        }
+
+        Uri primaryUri = Uri.create("primary");
+        Uri secondaryUri1 = Uri.create("secondary1");
+        Uri secondaryUri2 = Uri.create("secondary2");
+        Uri secondaryUri3 = Uri.create("secondary3");
+
+        AddressSelectorWrapper addressSelectorWrapper = AddressSelectorWrapper.Builder.Simple.create()
+            .withPrimary(primaryUri)
+            .withSecondary(ImmutableList.of(secondaryUri1, secondaryUri2, secondaryUri3))
+            .build();
+        sessionContainer = Mockito.mock(ISessionContainer.class);
+        IAuthorizationTokenProvider authorizationTokenProvider = Mockito.mock(IAuthorizationTokenProvider.class);
+        serviceConfigReader = Mockito.mock(GatewayServiceConfigurationReader.class);
+
+        consistencyWriter = new ConsistencyWriter(clientContext,
+            addressSelectorWrapper.addressSelector,
+            sessionContainer,
+            transportClientWrapper.transportClient,
+            authorizationTokenProvider,
+            serviceConfigReader,
+            false);
+
+        TimeoutHelper timeoutHelper = Mockito.mock(TimeoutHelper.class);
+        RxDocumentServiceRequest dsr = mockDocumentServiceRequest(clientContext);
+
+        consistencyWriter.writeAsync(dsr, timeoutHelper, false).subscribe();
+
+        String cosmosDiagnostics = dsr.requestContext.cosmosDiagnostics.toString();
+        assertThat(cosmosDiagnostics).containsOnlyOnce("storeResult");
```

Comment:
if two replica reach the diagnostic capture block at same time then we will have two results ?

Reply:
since for write scenario, we will only contacted the primary replica, so it will only contain one replica.
In the StoreReader test, since it could happen we will record multiple results, so the check there is >=1